qucs/qucs/qucs.cpp

```diff
@@ -1378,12 +1378,28 @@ bool QucsApp::gotoPage(const QString& Name)
   }
 
   QFileInfo Info(Name);
+  QString filename = Info.fileName();
+  QString icon = ":/bitmaps/empty.xpm";//Default icon (empty icon)
   if(Info.suffix() == "sch" || Info.suffix() == "dpl" ||
      Info.suffix() == "sym") {
-    d = DocumentTab->createEmptySchematic(Name);
-  } else {
-    d = DocumentTab->createEmptyTextDoc(Name);
+    d = new Schematic(this, Name);
+    if (d->isreadonly())
+    {// If the file is not writable, then add [READ-ONLY] to the tab title.
+     // The document can be edited but not saved.
+        filename += " [READ-ONLY]";
+    }
+    i = DocumentTab->addTab(dynamic_cast<Schematic*>(d), QPixmap(icon), filename);
+  }
+  else {
+    d = new TextDoc(this, Name);
+    if (d->isreadonly())
+    {// If the file is not writable, then add [READ-ONLY] to the tab title.
+     // The document can be edited but not saved.
+        filename += " [READ-ONLY]";
+    }
+    i = DocumentTab->addTab(dynamic_cast<TextDoc*>(d), QPixmap(icon), filename);
   }
+  DocumentTab->setCurrentIndex(i);
```

Comment:
Why all the changes above are not in `createEmptySchematic()` and `createEmptyTextDoc()`?